frontend/dispatcher.go

```diff
@@ -17,6 +17,47 @@ type HTTPError struct {
 	Message string
 }
 
+func db_requester() {
+	for {
+		db_request := <-db_open_channel
```

Comment:
@kyleterry 

> Use select instead and add the ability to shutdown the goroutine on sigterm or other signals.

The sigterm handler in the main function currently handles shutting down all of the currently open DBs. Is there a reason to shut down this goroutine?

> Also, lets avoid this global channel and maybe use a function that spawns a goroutine and returns a channel instead

I agree, global sucks. So in this case do I need to pass the new channel down to the handler functions that use it? Is there an alternative to just adding another argument to those functions?

> ...DB you potentially just opened close all the way when it makes it back to the top of the for/select cycle.

I'm a little confused on this one, if the database is already open it's simply returned from the global `databases` map. Could you elaborate on this one? Is the `select` suggestion for making this function handle closing of open database objects, in addition to opening them?

Thanks for the tips by the way. Not really sure what I'm doing in golang world, mostly just going off of Erlangisms.